src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java

```diff
@@ -592,4 +592,34 @@ void testGetCodeStatusIfCodeNotFoundInSvs() {
     CodeStatus status = vsacService.getCodeStatus(code, TEST_API_KEY);
     assertThat(status, is(equalTo(CodeStatus.NA)));
   }
+
+  @Test
+  void testUserUmlsLogout() {
+    when(umlsUserRepository.findByHarpId(anyString())).thenReturn(Optional.of(umlsUser));
+    UmlsUser user = vsacService.verifyUmlsAccess(TEST_API_KEY);
+    when(umlsUserRepository.deleteByHarpId(anyString())).thenReturn(Optional.of(umlsUser));
+    boolean loggedOut = vsacService.logoutUMLSUser(umlsUser.getHarpId());
+    assertThat(user.getHarpId(), is(equalTo(TEST_HARP_ID)));
+    assertThat(user.getApiKey(), is(equalTo(TEST_API_KEY)));
+    assertThat(loggedOut, is(equalTo(true)));
+  }
+
```

Comment:
We need another test case to verify when umlsUserRepository.deleteByHarpId(userName) returns Optional.empty() and fails to delete the entry

Reply:
@RohitKandimalla done